### PR TITLE
Orml xcm impl

### PR DESCRIPTION
**Pull request by Marky-Shi**
orml_xcm reference implementation

### CODE REVIEW

**Comment by Kailai-Wang** [comment]
- This branch is out-of-date with the base branch
- Do we have an issue with it? What's the background of this PR?

**Comment by Marky-Shi** [comment]
The relate issue link https://github.com/litentry/litentry-parachain/issues/725

**Comment by Kailai-Wang** [comment]
> The relate issue link #725

Thanks, let's use the [github keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) to refer to the issue link in the PR description so that the issue can get closed when the PR is merged.

File: runtime/common/src/tests/orml_xcm.rs [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+};
+use primitives::*;
+use xcm::latest::prelude::*;
+use xcm_executor::traits::Convert;
+
+pub fn orml_xcm_root_works<
+	Origin: OriginTrait
+		+ From<RawOrigin<AccountId>>
+		+ Clone
+		+ std::convert::From<
```
Comment (wangminqi) [inline runtime/common/src/tests/orml_xcm.rs:38]: This restriction of convert pallet_balances::Instance** is suspicious. Please check and feedback the reason for doing so.

File: runtime/common/src/tests/orml_xcm.rs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+			pallet_collective::RawOrigin<sp_runtime::AccountId32, pallet_balances::Instance1>,
+		> + std::fmt::Debug,
```
Comment (wangminqi) [inline runtime/common/src/tests/orml_xcm.rs]: same above. fmt restriction is also weird to display in the code.

File: runtime/rococo/src/xcm_config.rs [hunk 3]
```diff
@@ -278,14 +285,57 @@ parameter_types! {
 	pub const MaxAssetsForTransfer: usize = 3;
 }
 
+/// `EnsureOrigin` barrier to convert from dispatch origin to XCM origin, if one exists.
+pub struct EnsureXcmOriginWithSpecialRootGroup<Origin, Conversion, SpecialRootGroup>(
```
Comment (wangminqi) [inline runtime/rococo/src/xcm_config.rs]: Please replace Litentry part to rococo too.

**Review by wangminqi — CHANGES_REQUESTED** [review]

**Review by wangminqi — APPROVED** [review]

**Pull request merged**